client/scripts/app.ts

```diff
@@ -137,28 +137,36 @@ export function handleUpdateEmailConfirmation() {
   }
 }
 
-export async function selectCommunity(c?: CommunityInfo): Promise<void> {
+export async function selectCommunity(c?: CommunityInfo): Promise<boolean> {
   // Check for valid community selection, and that we need to switch
   if (app.community && c === app.community.meta) return;
 
   // Shut down old chain if applicable
-  const oldCommunity = app.community && app.community.meta;
   await deinitChainOrCommunity();
 
-  // Initialize the community
-  app.community = new Community(c, app);
-  await app.community.init();
+  // Begin initializing the community
+  const newCommunity = new Community(c, app);
+  const finalizeInitialization = await newCommunity.init();
```

Comment:
Can we explain what's happening here (and in the other place where `finalizeInitialization` is used as a variable?)

Also, finalizeInitialization should probably be more specific as a variable name - it's not obvious what finalize means in this context.